src/ide/inlay_hints/mod.rs

```diff
@@ -39,72 +43,102 @@ pub fn inlay_hints(db: &AnalysisDatabase, params: InlayHintParams) -> Option<Vec
         .filter(|node| range.contains(node.span_without_trivia(db)))
         .filter_map(|node| StatementLet::cast(db, node))
     {
-        let Some(lookup_item) = db.find_lookup_item(let_statement.as_syntax_node()) else {
-            continue;
-        };
-        let Some(func) = lookup_item.function_with_body() else {
-            continue;
-        };
+        let pattern = let_statement.pattern(db);
 
-        let Some(body) = db.function_body(func).ok() else {
+        let Some(pattern_resultants) = db.get_node_resultants(pattern.as_syntax_node()) else {
             continue;
         };
 
-        let mut inference_data = lookup_item
-            .resolver_data(db)
-            .ok()?
-            .inference_data
-            .clone_with_inference_id(db, InferenceId::NoContext);
-        let mut inference = inference_data.inference(db);
+        for pattern_resultant in pattern_resultants
+            .iter()
+            .filter_map(|pattern_resultant| {
+                pattern_resultant.ancestors_with_self(db).find_map(|node| Pattern::cast(db, node))
+            })
+            .dedup()
```

Comment:
```suggestion
            .collect<OrderedHashSet<_>>()
```
dedup works on consecutive elements only